onnx/defs/tensor/defs.cc

```diff
@@ -2538,164 +2538,178 @@ ONNX_OPERATOR_SET_SCHEMA(
         .SetDoc(AffineGrid_ver20_doc)
         .FunctionBody(R"ONNX(
         {
-          int_zero = Constant <value_int: int=0> ()
-          int_four = Constant <value_int: int=4> ()
+          # naming one: 1, one_f: 1.0, one_v: [1], one_fv: [1.0]
+          one = Constant <value_int: int=1> ()
+          two = Constant <value_int: int=2> ()
+          zero = Constant <value_int: int=0> ()
+          four = Constant <value_int: int=4> ()
+          one_v = Constant <value_ints: ints = [1]> ()
+          zero_v = Constant <value_ints: ints = [0]> ()
+
+          minus_one = Constant <value_int: int=-1> ()
+          minus_one_f = CastLike (minus_one, theta)
+          zero_f = CastLike (zero, theta)
+          one_f = CastLike (one, theta)
+          two_f = CastLike (two, theta)
 
           constant_align_corners = Constant <value_int: int=@align_corners> ()
-          constant_align_corners_equal_zero = Equal (constant_align_corners, int_zero)
+          constant_align_corners_equal_zero = Equal (constant_align_corners, zero)
 
           size_ndim = Size (size)
-          condition_is_2d = Equal (size_ndim, int_four)
+          condition_is_2d = Equal (size_ndim, four)
 
-          grid = If (condition_is_2d) <
-              then_branch = g1 () => (grid_2d_then) { # => (float[N, H, W, 2])
-                  int_one = Constant <value_int: int=1> ()
-                  minus_one = Constant <value = float {-1.0}> ()
-                  zero = Constant <value = float {0.0}> ()
-                  one = Constant <value = float {1.0}> ()
-                  two = Constant <value = float {2.0}> ()
+          size_NCDHW = If (condition_is_2d) <
```

Comment:
I think this can be simplified a little bit by returning 5 different values (N, C, D, H, W) from the if-op. Basically, using
```
   N, C, D, H, W = If (condition_is_2d) <
```

Reply:
I updated the code according to the recommendation